fastmri_recon/data/test_generators.py

```diff
@@ -38,8 +41,53 @@ def flow_random_shapes(self,):
             mask_batch = mask_batch[..., 0]
             yield (kspaces, mask_batch), images
 
+    @threadsafe_generator
     def flow_z_filled_random_shapes(self,):
         random_shapes_gen = self.flow_random_shapes()
         for (kspaces, _), images in random_shapes_gen:
             z_filled = zero_filled_recon(kspaces[..., 0], crop=False)[..., None]
             yield z_filled, images
+
+class DataGenerator:
+    def __init__(self, af, data, batch_size=1):
+        self.af = af
+        self.data = data
+        self.size = data.shape[1]
```

Comment:
Here size is not very explicit since you only consider the first dimension of the image. We might have rectangular images soon. Also you probably don't need this since you have `im_shape` just under.